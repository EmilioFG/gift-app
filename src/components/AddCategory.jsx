import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  const handleChange = ({ target }) => setInputValue(target.value)

  return (
    <>
      <h3>Agregar Categor√≠a</h3>
      <form onSubmit={handleSubmit} aria-label="form">
        <input
          type="text"
          placeholder="Buscar Gif"
          value={inputValue}
          onChange={handleChange}
        />
      </form>
    </>
  )
}

AddCategory.prototypes = {
  onNewCategory: PropTypes.func.isRequired,
}
