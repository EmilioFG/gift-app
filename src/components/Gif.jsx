import PropTypes from 'prop-types';


export const Gif = ({ title, url}) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <h6>{title}</h6>
    </div>
  )
}

Gif.prototypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
