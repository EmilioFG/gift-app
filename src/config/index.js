export default {
  API_URL: process.env.VITE_APP_API_URL,
  API_KEY: process.env.VITE_APP_API_KEY,
}