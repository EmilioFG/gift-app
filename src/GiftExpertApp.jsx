import { useState } from 'react'

import { AddCategory, GifGrid } from './components';


export const GiftExpertApp = () => {

  const [categories, setCategories] = useState(['dragon ball z']);

  const handleAddCategory = (newCategory) => {
    const currentCategories = [...categories].map((category) => (category.toLowerCase()));
    if (currentCategories.includes(newCategory.toLowerCase()))
      return alert(`la categor√≠a ${newCategory} ya existe`);

    setCategories([...categories, newCategory]);
  };

  return (
    <>
      <h1>GiftExpertApp</h1>
      <AddCategory onNewCategory={handleAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  )
}