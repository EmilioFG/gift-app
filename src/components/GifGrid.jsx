import PropTypes from 'prop-types';

import { useGifs } from '../hooks/useGifs';
import { Gif } from './Gif';


export const GifGrid = ({ category }) => {

  const { gifs, isLoading } = useGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <span>...loading</span>}
      <div className='card-grid'>
          {gifs.map((gif) => (
            <Gif
              key={gif.id}
              {...gif}
            />
          ))}
        </div>
    </>
  )
}

GifGrid.prototypes = {
  category: PropTypes.string,
}
