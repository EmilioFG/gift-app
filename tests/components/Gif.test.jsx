import { render, screen } from '@testing-library/react';
import { Gif } from '../../src/components';


describe("Pruebas en <Gif />", () => {

  const title = 'zombie';
  const url = 'https://emiliogonzalez.dev';

  test("debe de hacer match con el snapchot", () => {
    const { container } = render(<Gif title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<Gif title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    const { src, alt } = screen.getByRole('img');

    // TODO: se hace esto para quitar un caracter adicional '/' que agregar src que se obtiene a travéz de screen
    expect(src.substring(0, src.length - 1)).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el título en el componente", () => {
    render(<Gif title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

});
