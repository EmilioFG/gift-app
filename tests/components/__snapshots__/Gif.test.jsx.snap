// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Gif /> debe de hacer match con el snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="zombie"
      src="https://emiliogonzalez.dev"
    />
    <h6>
      zombie
    </h6>
  </div>
</div>
`;
